scripts/tests/test_dropbox_migrate_to_external_accounts.py

```diff
@@ -0,0 +1,197 @@
+            external_account.oauth_key,
+            user_settings_document['access_token']
+        )
+        assert_equal(
+            external_account.display_name,
+            user_settings_document['dropbox_info']['display_name']
+        )
+
+    def test_make_new_user_settings(self):
+        user_settings_document = self.unlinked_user_settings[0]
+        user = User.load(user_settings_document['owner'])
+        user_settings = migration.make_new_user_settings(user)
+        user.reload()
+        assert_equal(
+            len(user._backrefs['addons']['dropboxusersettings']['owner']),
+            1
+        )
+        assert_equal(
+            user._backrefs['addons']['dropboxusersettings']['owner'][0],
+            user_settings._id
+        )
+        assert_false(hasattr(user_settings, 'access_token'))
+
+    def test_make_new_node_settings(self):
+        node_settings_document = self.node_settings_documents[0]
+        node = Node.load(node_settings_document['owner'])
+        user_settings_document = database['dropboxusersettings'].find_one({
+            '_id':  node_settings_document['user_settings']
+        })
+        external_account, user = migration.migrate_to_external_account(
+            user_settings_document
+        )
+        user_settings = migration.make_new_user_settings(user)
+        node_settings = migration.make_new_node_settings(
+            node,
+            node_settings_document,
+            external_account,
+            user_settings
+        )
+        assert_equal(
+            len(node._backrefs['addons']['dropboxnodesettings']['owner']),
+            1
+        )
+        assert_equal(
+            node._backrefs['addons']['dropboxnodesettings']['owner'][0],
+            node_settings._id
+        )
+
+    def test_remove_old_documents(self):
+        user_settings_collection = database['dropboxusersettings']
+        old_user_settings = list(user_settings_collection.find())
+        old_user_settings_count = user_settings_collection.count()
+        node_settings_collection = database['dropboxnodesettings']
+        old_node_settings = list(node_settings_collection.find())
+        old_node_settings_count = node_settings_collection.count
+        migration.migrate(dry_run=False, remove_old=False)
+        assert_equal(
+            database['dropboxusersettings'].count(),
+            15
+        )  # 3 + 3 + 3 + 6 (non-deleted)
+        assert_equal(
+            database['dropboxnodesettings'].count(),
+            12
+        )  # 3 + 3 + 3 + 3
+        migration.remove_old_documents(
+            old_user_settings, old_user_settings_count,
+            old_node_settings, old_node_settings_count,
+            dry_run=False
+        )
+        assert_equal(
+            database['dropboxusersettings'].count(),
+            6
+        )
+        assert_equal(
+            database['dropboxnodesettings'].count(),
+            6
+        )
+
+    def test_migrate(self):
```

Comment:
Can you split up this test to more explicitly test the different migration scenarios? For example, have a `test_migrate_deleted_dropbox_user_settings`, `test_migrate_unauthorized_node_settings_documents`, etc. That way if the tests fail it's easier to know why.